File: Java/Easy/RuntimeError/reformat-phone-number_6.java
class Solution {
    public String reformatNumber(String number) {
        
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < number.length(); i++) {
            if (Character.isDigit(number.charAt(i))) {
                sb.append(number.charAt(i));
            }
        }
        String digits = sb.toString();
        int n = digits.length();
        sb = new StringBuilder();
        for (int i = 0; i < n; i += 3) {
            if (i + 4 <= n) {
                sb.append(digits.substring(i, i + 3)).append('-');
            } else if (i + 2 == n) {
                sb.append(digits.substring(i, i + 2));
            } else {
                sb.append(digits.substring(i, i + 3));
            }
        }
        return sb.toString();
    }
}

File: C++/Easy/WrongAnswer/reformat-phone-number_6.cpp
class Solution {
public:
    string reformatNumber(string number) {
        
        string result = "";
        string temp = "";
        for (auto c : number) {
            if (c == ' ' || c == '-') {
                continue;
            }
            temp += c;
        }
        for (int i = 0; i < temp.size(); i += 3) {
            if (i + 4 == temp.size()) {
                result += temp.substr(i, 2) + "-" + temp.substr(i + 2, 2);
                break;
            }
            if (i + 3 == temp.size()) {
                result += temp.substr(i, 3);
                break;
            }
            result += temp.substr(i, 3) + "-";
        }
        return result;
    }
};